.github/workflows/lint-php.yml

```diff
@@ -0,0 +1,65 @@
+name: Lint PHP
+
+on:
+  workflow_call:
+    inputs:
+      PHP_MATRIX:
+        description: Matrix of PHP versions as a JSON formatted object.
+        default: '["8.0"]'
+        required: false
+        type: string
+      COMPOSER_ARGS:
+        description: Set of arguments passed to Composer.
+        default: '--prefer-dist'
+        required: false
+        type: string
+      LINT_ARGS:
+        description: Set of arguments passed to PHP Parallel Lint.
+        default: '. -e php --colors --show-deprecated'
+        required: false
+        type: string
+    secrets:
+      COMPOSER_AUTH_JSON:
+        description: Authentication for privately hosted packages and repositories as a JSON formatted object.
+        required: false
+      ENV_VARS:
+        description: Additional environment variables as a JSON formatted object.
+        required: false
+
+jobs:
+  lint-php:
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php-version: ${{ fromJson(inputs.PHP_MATRIX) }}
+    env:
+      COMPOSER_AUTH: '${{ secrets.COMPOSER_AUTH_JSON }}'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          tools: cs2pr, parallel-lint
+
+      - name: Set up custom environment variables
+        env:
+          ENV_VARS: ${{ secrets.ENV_VARS }}
+        if: ${{ env.ENV_VARS }}
+        uses: actions/github-script@v6
+        with:
+          script: |
+            JSON
+              .parse(process.env.ENV_VARS)
+              .forEach(envVar => core.exportVariable(envVar.name, envVar.value));
+
+      - name: Install Composer dependencies
+        uses: ramsey/composer-install@v1
+        with:
+          composer-options: ${{ inputs.COMPOSER_ARGS }}
+
+      - name: Run PHP lint check
+        run: parallel-lint ${{ inputs.LINT_ARGS }} --checkstyle | cs2pr
```

Comment:
shouldn't we move the `--checkstyle | cs2pr` to the `inputs.LINT_ARGS` ? @tyrann0us

Reply:
This would mean that all calling workflows that want to override the `LINT_ARGS` would have to manually take over the `cs2pr` part (and look up how that even works).
Besides, these are "meta arguments" that have nothing to do with the actual linting but only with what is done with the result.

Reply:
Fine for me. Was just asking for the sake of documenting it down somewhere ;)

Reply:
Extracted to #22.